lib/internal/Magento/Framework/Filesystem/Directory/DenyListPathValidator.php

```diff
@@ -71,7 +71,7 @@ public function validate(
 
         foreach ($this->fileDenyList as $file) {
             $baseName = pathinfo($actualPath, PATHINFO_BASENAME);
-            if (str_contains($baseName, $file) || preg_match('#' . "\." . $file . '#', $fullPath)) {
+            if (stripos($baseName, $file) !== false || preg_match('#' . "\." . $file . '#', $fullPath)) {
```

Comment:
As we already have the `symfony/polyfill-php80` in composer.lock file, I think it's better to add the `symfony/polyfill-php80` as a requirement to composer.json and use a new function.
https://github.com/magento/magento2/blob/9ae286d8a747f40fa0c89944f4aafc5cf4a5e9bb/composer.lock#L6645-L6727

If you still believe it should replaced with old function - please replace it with the equivalent:
`'' === $needle || false !== strpos($haystack, $needle);`
https://github.com/symfony/polyfill-php80/blob/1100343ed1a92e3a38f9ae122fc0eb21602547be/Php80.php#L93